shell/platform/android/io/flutter/embedding/android/FlutterActivity.java

```diff
@@ -389,6 +391,20 @@ public FlutterActivity() {
     lifecycle = new LifecycleRegistry(this);
   }
 
+  /**
+   * This method exists so that JVM tests can ensure that a delegate exists without
+   * putting this Activity through any lifecycle events, because JVM tests cannot handle
+   * executing any lifecycle methods, at the time of writing this.
```

Comment:
Should be possible. Does http://robolectric.org/androidx_test/ help any?

Reply:
I'm not sure, but we don't have support for the latest testing infra. I think we're still on robolectric v1, while robolectric is now on v3 or v4. If Flutter brings this testing infra completely up to date with the ecosystem then we can probably write a lot of better tests.